### PR TITLE
feat: refactor router transform implementation

### PR DESCRIPTION
**Pull request by saikumarrs**
## Description of the change

Refactored router transform implementation.
It was missing two things:
- Bugsnag notifications (both at event level error handler and the global one)
- Translation of the handled error to the newly added error types

To send errors to Bugsnag with rich metadata, we have changed the signature of `processRouterDest` to accept a new argument, `reqMetadata`.

Some other minor updates:
- `flowType` to `feature` name change in CDK v2 flow
- The default error type is changed to `TransformationError`. It is functionally same as the previous logic.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

N/A

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/versionedRouter.js [hunk 1]
```diff
@@ -181,12 +181,12 @@ async function compareWithCdkV2(destType, input, feature, v0Result, v0Time) {
   }
 }
 
-async function handleV0Destination(destHandler, destType, input, feature) {
+async function handleV0Destination(destHandler, destType, inputArr, feature) {
```
Comment (koladilip) [inline src/versionedRouter.js:184]: any reason for making this array?

Comment (saikumarrs) [reply src/versionedRouter.js:184]: As discussed, we are sending an additional argument to the `processRouterDest` method. Please see the PR description for more details.

File: src/v0/destinations/am/transform.js [hunk 2]
```diff
@@ -973,12 +989,22 @@ const processRouterDest = async inputs => {
...
           [input.metadata],
-          error.status || 400,
-          error.message || "Error occurred while processing payload.",
-          errRes.statTags
+          errObj.status,
+          errObj.message,
+          errObj.statTags
+        );
+
+        errNotificationClient.notify(
```
Comment (koladilip) [inline src/v0/destinations/am/transform.js:1000]: does it notify all the errors or some kind of filtering logic in implemented in `errNotificationClient.notify`

File: src/v0/util/index.js [hunk 3]
```diff
@@ -1643,7 +1645,25 @@ const checkInvalidRtTfEvents = (inputs, destType) => {
...
  * @returns
  */
-const simpleProcessRouterDest = async (inputs, destType, singleTfFunc) => {
+const simpleProcessRouterDest = async (
+  inputs,
+  destType,
+  singleTfFunc,
+  reqMetadata
+) => {
+  const getEventReqMetadata = event => {
```
Comment (koladilip) [inline src/v0/util/index.js:1654]: looks like this defined in multiple places, is it possible to reuse and avoid duplicate code?

File: src/v0/destinations/am/transform.js (hunk 2 above)
Comment (saikumarrs) [reply src/v0/destinations/am/transform.js:968]: Yes, it has filtering logic to ignore the known error types.

**Review by ujjwal-ab — APPROVED** [review]

**Review by mihir-4116 — APPROVED** [review]

File: src/v0/util/index.js (hunk 3 above)
Comment (saikumarrs) [reply src/v0/util/index.js]: Refactored.

**Review by koladilip — APPROVED** [review]

**Pull request merged**